### PR TITLE
feat: additional fee scalars

### PR DESCRIPTION
**Pull request by yuwen01**

### CODE REVIEW

File: specs/protocol/holocene/exec-engine.md [hunk 1]
```diff
@@ -73,3 +73,64 @@ directly store this information.
 
 [l2-to-l1-mp]: ../../protocol/predeploys.md#L2ToL1MessagePasser
 [output-root]: ../../glossary.md#l2-output-root
+
+## Fees
+
+Holocene adds 2 new components to the fee calculation: the `gas_premium_fee` and the `constant_fee`.
```
Comment (puma314) [inline specs/protocol/holocene/exec-engine.md]: Holocene adds a new component to the fee calculation we call the `ConfigurableFee` parametrized by 2 scalars the `configurableFeeConstant` and `configurableFeeScalar`.

File: specs/protocol/holocene/exec-engine.md [hunk 2]
```diff
@@ -73,3 +73,64 @@ directly store this information.
 
 [l2-to-l1-mp]: ../../protocol/predeploys.md#L2ToL1MessagePasser
 [output-root]: ../../glossary.md#l2-output-root
+
+## Fees
+
+Holocene adds 2 new components to the fee calculation: the `gas_premium_fee` and the `constant_fee`. 
+New OP stack variants have different resource consumption patterns, and thus require a more
```
Comment (puma314) [inline specs/protocol/holocene/exec-engine.md]: , "which this configurableFee" enables.

File: specs/protocol/holocene/exec-engine.md [hunk 3]
```diff
@@ -73,3 +73,64 @@ directly store this information.
...
+
+## Fees
+
+Holocene adds 2 new components to the fee calculation: the `gas_premium_fee` and the `constant_fee`. 
+New OP stack variants have different resource consumption patterns, and thus require a more 
+flexible pricing model.
+
+### Fee Vaults
+
+In addition to the existing 3 fee vaults (The [`SequencerFeeVault`][sequencer-fee-vault]
```
Comment (puma314) [inline specs/protocol/holocene/exec-engine.md]: Just 1 vault -- configurable fee vault.

File: specs/protocol/holocene/predeploys.md [hunk 4]
```diff
@@ -228,6 +242,11 @@ A new function is added to fetch the full Fee Vault Config.
 function config()(address,uint256,WithdrawalNetwork)
 ```
 
+### ConfigurableFeeVault
+
```
Comment (tynes) [inline specs/protocol/holocene/predeploys.md]: Why add a new fee vault? This wasn't discussed in the design doc at all

File: specs/protocol/holocene/exec-engine.md [hunk 5]
```diff
@@ -73,3 +77,50 @@ directly store this information.
...
+
+`configurableFeeScalar` and `configurableFeeConstant` are loaded in a similar way to the `baseFeeScalar` and
+`blobBaseFeeScalar` used in the [`L1Fee`](../../protocol/exec-engine.md#ecotone-l1-cost-fee-changes-eip-4844-da).
+calculation. In more detail, these scalars can be accessed in two interchangable ways.
+
+- read from the deposited L1 attributes (`configurableFeeScalar` and `configurableFeeConstant`) of the current L2 block
+- read from the L1 Block Info contract (`0x4200000000000000000000000000000000000015`)
+  - using the respective solidity `uint64`-getter functions (`configurableFeeScalar`, `configurableFeeConstant`)
+  - using direct storage-reads:
+    - Configurable fee scalar as big-endian `uint64` in slot `7`
```
Comment (tynes) [inline specs/protocol/holocene/exec-engine.md]: I would recommend storage packing these

**Comment by tynes** [comment]
I recommend calling this something else rather than prover gas. It should be something more about fee formula

File: specs/protocol/isthmus/l1-attributes.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+| basefee           | uint256 | 36-67          | 2       |
+| blobBaseFee       | uint256 | 68-99          | 3       |
+| l1BlockHash       | bytes32 | 100-131        | 4       |
+| batcherHash       | bytes32 | 132-163        | 5       |
+| eip1559Denominator   | uint64  | 164-171        | 6       |
+| eip1559Elasticity    | uint64  | 172-179        |         |
+| configurableFeeScalar  | uint64  | 180-187        |         |
+| configurableFeeConstant    | uint64  | 188-195        |         |
+
+In the first L2 block after the Holocene activation block, the Holocene L1 attributes are first used.
```
Comment (puma314) [inline specs/protocol/isthmus/l1-attributes.md]: Holocene -> Isthmus

**Review by puma314 — APPROVED** [review]

**Pull request closed without merging**